data/frame_json.go

```diff
@@ -57,23 +56,113 @@ func (codec *dataFrameCodec) Decode(ptr unsafe.Pointer, iter *jsoniter.Iterator)
 	*((*Frame)(ptr)) = frame
 }
 
-// FrameToJSON writes a frame to JSON.
-// NOTE: the format should be considered experimental until grafana 8 is released.
-func FrameToJSON(frame *Frame, includeSchema bool, includeData bool) ([]byte, error) {
+// FrameJSONInclude - Custom type to hold value for weekday ranging from 1-7
+type FrameJSONInclude int
+
+// Declare related constants for each weekday starting with index 1
+const (
+	SchemaAndData FrameJSONInclude = iota + 1 // EnumIndex = 1
+	OnlyData                                  // EnumIndex = 2
+	OnlySchema                                // EnumIndex = 3
+)
+
+// FrameJSON holds a byte representation of the schema separate from the data
+type FrameJSON struct {
+	schema json.RawMessage
+	data   json.RawMessage
+}
+
+// Body returns the bytes to both schema and data (if they exist)
+func (f *FrameJSON) Bytes(args FrameJSONInclude) []byte {
```

Comment:
A bit worried about this scenario:
```go
frameJSON, _ := data.FrameToJSON(frame, IncludeSchemaOnly)
// And later:
jsonBytes := frameJSON.Bytes(IncludeDataOnly)
```

Reply:
any ideas to make it better?  I added a comments, but not sure if there is a better way

Reply:
I removed the option to only create schema or data -- it now always makes both and you specify what you want when you get the bytes